### PR TITLE
fix(addon/karpenter): Disable Interruption Queue on Helm when the addon flag is false. 

### PR DESCRIPTION
**Pull request by youngjeong46**
*Issue #, if available:* #799

*Description of changes:* Interruption Queue Name was resolved to the stack name even though the queue was disabled. This PR will fix to ensure if the queue is disabled, then the helm interruption queue name will resolve to a blank value (as intended for default helm value).

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: lib/addons/karpenter/index.ts [hunk 1]
```diff
@@ -280,7 +280,7 @@ export class KarpenterAddOn extends HelmAddOn {
                 clusterEndpoint: endpoint,
                 clusterName: name,
                 defaultInstanceProfile: karpenterInstanceProfile.instanceProfileName,
-                interruptionQueueName: stackName
+                interruptionQueueName: interruption ? stackName : "",
```
Comment (shapirov103) [inline lib/addons/karpenter/index.ts:283]: If interruptionQueueName is empty string, is the same as not providing the interruptionQueueName attribute at all? Was it tested?

**Review by shapirov103 — COMMENTED** [review]
One question, looks good otherwise.

File: lib/addons/karpenter/index.ts (hunk 1 above)
Comment (youngjeong46) [reply lib/addons/karpenter/index.ts:283]: yes - the default value is "" as shown [here](https://github.com/aws/karpenter/blob/5e02e4c6eb3432b5da12df100474a114db66607f/charts/karpenter/values.yaml#L187)

**Review by shapirov103 — APPROVED** [review]
LGTM

**Pull request merged**